projects/ngx-drag-to-select/src/lib/select-container.spec.ts

```diff
@@ -47,12 +60,117 @@ describe('SelectContainerComponent', () => {
   it('should expose update as part of the public api', () => {
     const selectContainer = fixture.debugElement.query(By.directive(SelectContainerComponent));
 
-    jest.spyOn(selectContainer.componentInstance, 'calculateBoundingClientRect');
+    jest.spyOn(selectContainer.componentInstance, '_calculateBoundingClientRect');
     jest.spyOn(fixture.componentInstance.selectItem, 'calculateBoundingClientRect');
 
     selectContainer.componentInstance.update();
 
-    expect(selectContainer.componentInstance.calculateBoundingClientRect).toHaveBeenCalled();
+    expect(selectContainer.componentInstance._calculateBoundingClientRect).toHaveBeenCalled();
     expect(fixture.componentInstance.selectItem.calculateBoundingClientRect).toHaveBeenCalled();
   });
+
+  describe('selectItems()', () => {
+    it('should select items', done => {
+      const ids = [1, 2];
+      const result = [{ id: 1 }, { id: 2 }];
+
+      selectContainerInstance.select.subscribe(items => {
+        expect(componentInstance.selectedItems.length).toBe(result.length);
+        expect(items).toEqual(result);
+        expect(componentInstance.selectedItems).toEqual(result);
+        done();
+      });
+
+      selectContainerInstance.selectItems((item: SelectItemValue) => ids.includes(item.id));
+    });
+
+    it('should not throw error when selecting items that do not exist', done => {
+      const result = [];
+
+      selectContainerInstance.select.subscribe(items => {
+        expect(componentInstance.selectedItems.length).toBe(result.length);
+        expect(items).toEqual(result);
+        expect(componentInstance.selectedItems).toEqual(result);
+        done();
+      });
+
+      selectContainerInstance.selectItems((item: SelectItemValue) => item.id === -1);
+      selectContainerInstance.selectItems((item: SelectItemValue) => item.id === 100);
+    });
+  });
+
+  describe('deselectItems()', () => {
+    beforeEach(() => {
+      selectContainerInstance.selectItems((item: SelectItemValue) => [1, 2, 3].includes(item.id));
```

Comment:
I think this will be clearer if we just return `true` instead of `[1, 2, 3].includes(item.id)`.

Reply:
Great remark! Makes sense.